keps/sig-api-machinery/1027-api-unions/README.md

```diff
@@ -86,42 +121,141 @@ become possible.
 
 ### Goals
 
-The goal is to enable a union or "oneof" semantics in Kubernetes types, both for
-in-tree types and for CRDs.
+Provide a simple mechanism for API authors to label fields of a resource as
+members of a oneOf, in order to receive standardized validation and
+normalization, rather than having to author it themeselves per
+resource as currently done as a workaround in various validation
+functions (e.g. `pkg/apis/<group>/validation/validation.go`).
+
+* Validation - ensuring only one member field is set (or at most one if
+  desired).
+* Normalization - ensuring the API server can modify the fields of the oneOf to
+  best match the intent of the client, despite the client potentially having
+  incomplete information
+
+Another goal is to migrate at least one existing union to the new semantics in
```

Comment:
I'm  not completely sure of this at the moment.  I might be willing to accept a "from now on" state without requiring a fix to previous mistakes.

Reply:
Removed `this` and added a bit to the testing and graduation criteria based on Daniel's soft acceptance of testing as sufficient for validating the possibility of migrations.